### PR TITLE
Geopy 235

### PR DESCRIPTION
**Pull request by benk-mira**
**GEOPY-235 - Support grav inversion for simpeg/tiled-simulation** 
 Initial design for a gravity driver.  Will serve as a good template to start a base driver class.

### CODE REVIEW

File: geoapps/io/Gravity/constants.py [hunk 1]
```diff
@@ -0,0 +1,1021 @@
...
+        "default": None,
+        "enabled": False,
+        "value": None,
+    },
+    "geoh5": {
+        "default": None,
+        "enabled": False,
+        "value": None,
+    },
+    "run_command": {
```
Comment (domfournier) [inline geoapps/io/Gravity/constants.py:763]: This should be changed to "run_command": "geoapps.drivers.grav_inversion"

File: geoapps/io/Gravity/constants.py [hunk 2]
```diff
@@ -0,0 +1,1021 @@
...
+        "enabled": False,
+        "value": "geoapps.drivers.grav_inversion",
+    },
+    "run_command_boolean": {
+        "default": False,
+        "value": False,
+        "label": "Run python module ",
+        "tooltip": "Warning: launches process to run python model on save",
+        "main": True,
+    },
```
Comment (domfournier) [inline geoapps/io/Gravity/constants.py:774]: Add one more for:
`"conda_environment": "geoapps"`

**Review by domfournier — COMMENTED** [review]
One small step for man, one giant leap for inversion

**Comment by domfournier** [comment]
Tried to run it from a default ui.json written to disk. I am getting an error with the window?
image

Here is the ui.json
[Grav_inversion.ui.zip](https://github.com/MiraGeoscience/geoapps/files/6712517/Grav_inversion.ui.zip)

File: geoapps/io/Gravity/constants.py (hunk 1 above)
Comment (benk-mira) [reply geoapps/io/Gravity/constants.py:763]: yep!

File: geoapps/io/Gravity/constants.py (hunk 2 above)
Comment (benk-mira) [reply geoapps/io/Gravity/constants.py:774]: done!

File: geoapps/io/Gravity/constants.py (hunk 1 above)
Comment (domfournier) [reply geoapps/io/Gravity/constants.py:763]: This doesn't work. The `run_command` and `conda_environment` can't be dictionaries. Just a string.

File: geoapps/io/Gravity/constants.py (hunk 2 above)
Comment (domfournier) [reply geoapps/io/Gravity/constants.py:774]: Same here

File: geoapps/io/Gravity/constants.py [hunk 3]
```diff
@@ -0,0 +1,1022 @@
...
+            "{48F5054A-1C5C-4CA4-9048-80F36DC60A06}",
+        ],
+        "value": None,
+    },
+    "reference_model": {
+        "association": "Cell",
+        "dataType": "Float",
+        "default": 0.0,
+        "group": "Models",
+        "isValue": False,
```
Comment (domfournier) [inline geoapps/io/Gravity/constants.py:614]: Default should be True

File: geoapps/io/Gravity/constants.py [hunk 4]
```diff
@@ -0,0 +1,1022 @@
...
+        ],
+        "label": "starting model object",
+        "value": None,
+    },
+    "starting_model": {
+        "association": "Cell",
+        "dataType": "Float",
+        "default": None,
+        "group": "Starting Model",
+        "isValue": False,
```
Comment (domfournier) [inline geoapps/io/Gravity/constants.py:119]: Inversion crashes if the user doesn't click on value. Should default to "isValue"=true, with default value.

**Review by domfournier — COMMENTED** [review]
See changes below. Somehow the parallelization doesn't seem to kick in... but we can fix that later.

File: geoapps/io/Gravity/constants.py (hunk 3 above)
Comment (benk-mira) [reply geoapps/io/Gravity/constants.py:614]: isValue should be True!

File: geoapps/io/Gravity/constants.py (hunk 4 above)
Comment (benk-mira) [reply geoapps/io/Gravity/constants.py:119]: Gotcha, fixed up.

**Review by domfournier — APPROVED** [review]
Fine by me. Let's fix up the warnings on GEOPY-268

**Pull request merged**